addons/point_of_sale/models/pos_order.py

```diff
@@ -835,27 +844,34 @@ def create_picking(self):
                 })
 
             # prefer associating the regular order picking, not the return
-            order.write({'picking_id': order_picking.id or return_picking.id})
-
-            if return_picking:
-                order._force_picking_done(return_picking)
-            if order_picking:
-                order._force_picking_done(order_picking)
+            order.write({'picking_id': new_picking.id or new_return.id})
 
             # when the pos.config has no picking_type_id set only the moves will be created
-            if moves and not return_picking and not order_picking:
+            if moves and not order.picking_id:
                 moves._action_assign()
                 moves.filtered(lambda m: m.product_id.tracking == 'none')._action_done()
 
+        if return_picking:
+            self.env['pos.order']._force_picking_done(return_picking)
+        if order_picking:
+            self.env['pos.order']._force_picking_done(order_picking)
+
         return True
 
-    def _force_picking_done(self, picking):
+    def _force_picking_done(self, picking_dict):
         """Force picking in order to be set as done."""
-        self.ensure_one()
-        picking.action_assign()
-        wrong_lots = self.set_pack_operation_lot(picking)
-        if not wrong_lots:
-            picking.action_done()
+        pickings_to_assign = self.env['stock.picking']
+        for order_id, picking in picking_dict.items():
```

Comment:
do we really need to assign before done, or can we imediattely done?
If we can if not wrong_lots -> action_done, else -> action assign

If we need to assign, remove the loop and do picking_to_assign = self.env['stock.picking'].concat(*picking_dict.values())